### PR TITLE
Failed requests for URLs containing <no source> (Issue)

### LINKED ISSUE #1053: Failed requests for URLs containing <no source>
### Expected behaviour
No requests for URLs containing `<no source>`

### Actual behaviour
Logs show failed requests to `/s/css/census/themes/census/<no%20source>`

### Steps to reproduce the behaviour
Unknown, these errors are from the server logs

### PR DESCRIPTION
**Pull request by adoublebarrel**
### What is the context of this PR?
[Trello Card](https://trello.com/c/SX3FWZrm/1025-eq-survey-runner-1053-failed-requests-for-urls-containing-no-source-issue-xs)

Fixes https://github.com/ONSdigital/eq-survey-runner/issues/1053

After some poking around (and upgrading the gulp sass plugin) it looks
like a problem with the globs + postCSS + cssnano was resulting in a `themes/census/<no
source>` path being added to the gulp styles task. Seems the strongest candidate for the root cause so I've fixed it.

Minified assets with source maps are now generated for all environments.

### How to review
You can run the app locally as usual and everything is fine.

- You'll get minified versions of the CSS and JS
- You should see a comment at the end of the minified files detailing the source map url.
- In chrome you can search for the mapped files using CMD-P (CTRL-P for non-OSX)

Run `EQ_MINIMIZE_ASSETS=False yarn compile`
Run `EQ_MINIMIZE_ASSETS=False ./scripts/run_app.sh` from runners root dir

- All surveys should look the same
- Viewing the network traffic in dev tools/reviewing the server log should show non minified versions of the CSS and JS are being used.

### CODE REVIEW

File: gulp/styles.js [hunk 1]
```diff
@@ -35,21 +35,31 @@ export function lint() {
 
 export function styles() {
   const minifyAssets = process.env.EQ_MINIMIZE_ASSETS === 'True'
+  const doSourceMaps = process.env.EQ_DEV_MODE === 'True'
```
Comment (jonnyshaw89) [inline gulp/styles.js]: We are trying to stop the use of the generic "Dev Mode" so maybe if we want this feature it should have its own switch?

File: app/settings.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def get_env_or_fail(key):
     return value
 
 
-EQ_MINIMIZE_ASSETS = parse_mode(os.getenv('EQ_MINIMIZE_ASSETS', 'False'))
+EQ_MINIMIZE_ASSETS = parse_mode(os.getenv('EQ_MINIMIZE_ASSETS', 'TRUE'))
```
Comment (ajmaddaford) [inline app/settings.py]: This should be 'True' to be consistent with the other `parse_mode` calls

**Review by hamishtaplin — APPROVED** [review]

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -31,6 +31,7 @@ services:
       EQ_ENABLE_CACHE: "True"
       EQ_DEVELOPER_LOGGING: "True"
       EQ_RABBITMQ_ENABLED: "True"
+      EQ_MINIMIZE_ASSETS: "True"
```
Comment (jonnyshaw89) [inline docker-compose.yml]: No need to set this to True if the app defaults to True

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/settings.py (hunk 2 above)
Comment (adoublebarrel) [reply app/settings.py]: 🤦‍♂️

File: scripts/dev_settings.sh [hunk 4]
```diff
@@ -9,7 +9,7 @@ if [ -z "$EQ_RABBITMQ_ENABLED" ]; then
 fi
 
 if [ -z "$EQ_MINIMIZE_ASSETS" ]; then
-  export EQ_MINIMIZE_ASSETS=False
+    export EQ_MINIMIZE_ASSETS=True
```
Comment (jonnyshaw89) [inline scripts/dev_settings.sh]: If the `settings.py` defaults this to on then there is no need to also set it here.

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

**Review by jonnyshaw89 — APPROVED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@adoublebarrel squash?

**Pull request merged**